src/commands/NFD.ts

```diff
@@ -851,6 +846,41 @@ class NFD {
       })
   }
 
+  private async makeCollage(nfdList: NFDItem[]) {
+    const workingList = nfdList.slice(0, Math.min(nfdList.length, this.MAX_COLLAGE_ITEMS))
+
+    const columnCount = Math.ceil(Math.sqrt(workingList.length))
+    const rowCount = Math.ceil(workingList.length / columnCount)
+
+    const collageWidth = columnCount * this.NFD_WIDTH + (columnCount - 1) * this.COLLAGE_COLLUMN_MARGIN
+    const collageHeight = rowCount * this.NFD_HEIGHT + (rowCount - 1) * this.COLLAGE_ROW_MARGIN
+
+    const compositeList: sharp.OverlayOptions[] = []
+
+    for (let i = 0; i < workingList.length; i++) {
```

Comment:
I think this could be cleaner if you use `map` on the workingList.

Reply:
I need the index number i to work out the position in the collage to blit to. Unless I am misunderstanding?